app/controllers/courses/sites_controller.rb

```diff
@@ -1,10 +1,19 @@
 module Courses
   class SitesController < ApplicationController
     decorates_assigned :course
-    before_action(
-      :build_course,
-      :build_provider,
-    )
+    before_action :build_course_params, only: %i[continue]
+
+    include CourseBasicDetailConcern
+    before_action :build_course, only: %i[edit update]
+    before_action :build_provider_with_sites
+
+    def new
+      if @provider.sites.count == 1
+        params["course"] ||= {}
+        params["course"]["sites_ids"] = [@provider.sites.first.id]
```

Comment:
Might be good to move these two lines to a function called `set_default_site` or something